tests/hmm/hmumu_utils.py

```diff
@@ -1144,17 +1161,33 @@ def fix_muon_fsrphoton_index(offsets_fsrphotons, offsets_muons, fsrphotons_dROve
             #Index of FSR photon associated to muon
             fidx = offsets_fsrphotons[iev] + muons_fsrPhotonIdx[midx]
 
-            if k != muons_fsrPhotonIdx[midx]:
-                if fsrphotons_dROverEt2[i] < fsrphotons_dROverEt2[fidx]:
-                    out_muons_fsrPhotonIdx[midx] = k
+            px = mu_pt[midx] * np.cos(mu_phi[midx])
```

Comment:
could also use`hepaccelerate.spherical_to_cartesian` here: https://github.com/hepaccelerate/hepaccelerate/blob/master/hepaccelerate/kernels.py#L4

Reply:
Thank you for pointing out this info! will improve in the next PR.